django/thunderstore/api/cyberstorm/views/team.py

```diff
@@ -52,6 +62,58 @@ def get_queryset(self) -> QuerySet[TeamMember]:
         )
 
 
+class CyberstormTeamAddMemberRequestSerialiazer(serializers.Serializer):
+    user = serializers.CharField(max_length=User._meta.get_field("username").max_length)
+    role = serializers.ChoiceField(choices=TeamMember._meta.get_field("role").choices)
+
+
+class CyberstormTeamAddMemberResponseSerialiazer(serializers.Serializer):
+    user = serializers.CharField(max_length=User._meta.get_field("username").max_length)
+    role = serializers.ChoiceField(choices=TeamMember._meta.get_field("role").choices)
+    team = serializers.CharField(max_length=Team._meta.get_field("name").max_length)
```

Comment:
You don't need to add validation rules to the response serializer, there's no reason we need to validate it for here. The only reason you might want to do that is so they're built into the API documentation, but in that case we don't want to be over-specific since that could promote bad API client design and lock us in to those choices. For example, there's no guarantee that the current roles are the only roles that the API is going to communicate in the future, and as such we shouldn't include it in the documentation.

In other words, you should get rid of the constraints here entirely

Reply:
How about the role choices? I'd keep that restriction there to make it clear for client on what's the choices to come out.

Let me know if you disagree 🤷‍♂️